### PR TITLE
feat: Pivot Points segments and price-channel layering

### LINKED ISSUE #498: chart API: expose the remaining unchartable indicators (Renko, Correlation, PRS) so their docs pages can chart
## Summary

Several indicator documentation pages on the Stock Indicators for .NET docs site cannot render an interactive `<StockIndicatorChart>` because the indicator has **no entry in the chart API catalog** (`GET /indicators`) and therefore no data endpoint. Please consider adding catalog listings + endpoints so these pages can show charts.

## Context

- Docs consume `@facioquo/indy-charts` (Vue) + the chart API at `charts-api.stockindicators.dev`. (Was `stock-charts-api.azurewebsites.net` when this issue was filed; the backend moved to Cloudflare in #539.)
- The component renders only indicators present in the catalog returned by `GET /indicators` (each needs a `uiid`, `endpoint`, `chartType`, `parameters`, `results`).
- Today the catalog has 83 indicators. The pages below map to indicators that are absent, so they're the only non-category pages with no chart.

## Requested additions (by difficulty)

**Straightforward — derived from an existing endpoint / single series:**
- [x] **Gator Oscillator** — derived from `ALLIGATOR` (jaw/teeth/lips). Two histograms (upper = |jaw − teeth|, lower = −|teeth − lips|), `chartType: oscillator`. Could reuse the `ALLIGATOR` calc/endpoint and expose `upper`/`lower` results, mirroring how `BB` exposes both `BB` (overlay) and `BB-PCTB` (oscillator). Implemented in #499 
- [x] **Heikin-Ashi** — `chartType: overlay` (candlestick transform; open/high/low/close). Implemented in #535 + #536 (candle-rendered overlay that replaces the price candles); published in `@facioquo/indy-charts@0.9.0`.
- [ ] **Renko** — `chartType: overlay` (brick transform).
- [x] **Quote transforms (HL2, HLC3, OC2, OHL3, OHLC4 / "QuotePart")** — `chartType: overlay`, a single derived price line. Implemented in #499 
- [x] **Pivot Points / Rolling Pivot Points** — `chartType: overlay` (support/resistance levels). Partially implemented in #499 
- [x] **Williams Fractal "Pivots"** — `chartType: overlay` (point markers; `FRACTAL` already exists, but the "Pivots" page has no listing). Implemented in #499 
- [x] **SMA analysis (MAD / MAPE / MSE)** — `chartType: oscillator` (error metrics off an SMA). Re-implemented in #534 as per-metric listings (`SMA-MAD` / `SMA-MAPE` / `SMA-MSE`) after the original single-chart version was removed for mixed-unit unreadability; closes #475.

**Needs a benchmark/second series (API shape question):**
- [ ] **Correlation Coefficient / R²** and **Price Relative Strength (PRS)** — both require a comparison symbol. The current single-series endpoint shape doesn't express "indicator vs. benchmark," so these may need a new parameter (e.g. a second symbol) before they're chartable.

## Pages currently chart-less for this reason

`Renko`, `Correlation`, `Prs`.

Originally also `Gator`, `HeikinAshi`, `QuotePart`, `PivotPoints`, `RollingPivots`, `Pivots`, and `SmaAnalysis` — all now charted via #499, #534, #535, and #536.

Consolidates (closed as duplicate of this issue):
- #476 — its Pivot Points / Pivots / Rolling Pivots portion shipped in #499 + #500; its remaining items (Correlation, PRS, Heikin-Ashi) are tracked above.

## Notes

- For anything that's genuinely not chartable (e.g. needs a benchmark), a clear "not charted" decision is fine — this issue is to track which ones *can* be exposed.
- Once a listing exists, wiring the docs page is a one-line `<StockIndicatorChart indicator="…" />` (plus a registry entry), so the docs side is cheap.

### PR DESCRIPTION
**Pull request by DaveSkender**
## Summary

Fixes the rendering of the Pivot Points overlay added in #499 and makes price-channel overlays layer correctly behind the price.

- **Monthly Pivot Points render as discrete per-window segments.** The levels are constant within a window and step at each boundary; previously they drew as one continuous diagonal staircase. Now each window is its own flat horizontal segment, matching the [reference](https://dotnet.stockindicators.dev/indicators/PivotPoints).
- **Restyled** to thin lines (`LineWidth 1`) with a gray pivot line (`DarkGray`), consistent with other band overlays.
- **Price-channel overlays now render behind the price candles** via the listing `Order` field, which previously had no effect.

## How the segment break works

Two approaches don't work here: a Chart.js per-segment `borderColor` callback is a *function*, which breaks the `structuredClone` resize cache; and inserting weekend gap points breaks the index-based window slicing (`data.slice(startIndex)`). Instead, `buildDataPoints` **replaces** the first value of each new window with `NaN`. Chart.js doesn't span gaps by default, so the line breaks per window, and replacing (not inserting) keeps the dataset index-aligned with quotes.

A declarative `Segmented` result flag opts in only the weekly/monthly pivot levels. Rolling Pivots stays continuous (its rolling window recomputes every bar).

## Z-order wiring

`dataset.order` was sourced from a per-result `order` that the API never emits, so every overlay drew at Chart.js default `0` — *in front* of the price (candles 75, volume 76) — and the listing-level `Order` enum (e.g. `BehindPrice`) had no effect. `createDefaultSelection` now uses `config.order ?? listing.order`, so the listing order drives draw order as intended. This brings all 8 `price-channel` overlays behind the price consistently (Bollinger, Donchian, Keltner, MA Envelopes, Pivot Points, Rolling Pivots, StdDev Channels, STARC). The result-level `order` is now optional in the type to reflect that it isn't emitted today.

## Changes

- `Endpoints.cs` — `ToPivotPoints(PeriodSize.Month)`
- `IndicatorResultConfig.cs` — `Segmented` flag (`JsonIgnore` when default)
- `Service.Metadata.cs` — Pivot Points: `Segmented`, thin lines, gray PP, `BehindPrice`; Rolling Pivots: `BehindPrice`
- `data/transformers.ts` — null the first value of each window for segmented results
- `helpers/create-default-selection.ts` — `order: config.order ?? listing.order`
- `config/types.ts` — `segmented?`, optional `order?`
- `backup-indicators.json` — snapshot refresh

## Testing

- Backend: Pivot Points levels are segmented / Rolling Pivots continuous (with explicit count guards)
- Frontend: window-boundary break, non-segmented lines unaffected, and edge cases (leading-null warmup, mid-series gap, consecutive steps)
- 188 frontend tests + backend tests pass; ESLint/tsc, Prettier, and `dotnet format` clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: server/WebApi.Tests/Endpoints/MainEndpointsTests.cs [hunk 1]
```diff
@@ -522,6 +522,30 @@ public async Task GetRollingPivots_WithValidParameters_ReturnsOkResult()
...
+    [Fact]
+    public void PivotPointsListing_MarksAllLevelsSegmented()
+    {
+        // weekly Pivot Points are piecewise-constant level lines: the client
+        // renders one horizontal segment per week, so every result opts in.
+        var listing = Metadata
+            .IndicatorListing("https://localhost")
+            .Single(l => l.Uiid == "PIVOT-POINTS");
+
+        Assert.All(listing.Results, r => Assert.True(r.Segmented));
```
[the remark this answers is not in the document]
Comment (DaveSkender) [reply server/WebApi.Tests/Endpoints/MainEndpointsTests.cs:537]: Addressed in f624e83 — both metadata tests now assert `Assert.Equal(7, listing.Results.Count)` before `Assert.All`, so they can't pass vacuously on an empty result set. Also applied the two nitpicks: documented that segmented breaking is for piecewise-constant series only, and added edge-case tests (leading-null warmup, mid-series gap, consecutive steps).
